src/odemis/util/img.py

```diff
@@ -811,3 +813,114 @@ def getBoundingBox(content):
+            integ_img = img
+            self._img = integ_img
+
+        else:
+            mda = img.metadata  # metadata of the new acquired image
+            mdb = self._img.metadata  # metadata of the previous acquired image or the previous integrated one
+            data = numpy.add(self._img, img, dtype=self._best_dtype)
+            # update the metadata of the integrated image in every integration step
+            md = self.add_integration_metadata(mda, mdb)
+
+            # At the end of the acquisition, check if the detector type is DT_NORMAL and then take the average by
+            # dividing with the number of acquired images (integration count) for every pixel position and restoring
+            # the original dtype.
+            if self._step == self.steps:
+                det_type = md.get(model.MD_DET_TYPE, model.MD_DT_INTEGRATING)
+                if det_type == model.MD_DT_NORMAL:  # SEM
+                    orig_dtype = img.dtype
+                    if orig_dtype.kind in "biu":
+                        data = numpy.floor_divide(data, self._step, dtype=orig_dtype)
+                    else:
+                        data = numpy.true_divide(data, self._step, dtype=orig_dtype, casting='unsafe')
+                if not det_type == model.MD_DT_INTEGRATING:  # optical
+                    logging.warning("Unknown detector type %s for image integration.", det_type)
+                # The baseline, if exists, should also be subtracted from the integrated image.
+                if model.MD_BASELINE in md:
+                    data, md = self.subtract_baseline(data, md)
+                logging.debug("Image integration is completed.")
+
+            integ_img = model.DataArray(data, md)
+            self._img = integ_img
+
+        # reset the ._img and ._step once you reach the integration count
+        if self._step >= self.steps:
+            self._step = 0
+            self._img = None
+
+        return integ_img
+
+    def add_integration_metadata(self, mda, mdb):
+        """
+        add mdb to mda, and update mda with the result
+        returns dict: mda, which has been updated
+        """
+        if MD_DWELL_TIME in mda:
+            mda[model.MD_DWELL_TIME] += mdb.get(model.MD_DWELL_TIME, 0)
+        if MD_EXP_TIME in mda:
+            mda[model.MD_EXP_TIME] += mdb.get(model.MD_EXP_TIME, 0)
+        mda[model.MD_INTEGRATION_COUNT] = mda.get(model.MD_INTEGRATION_COUNT, 1) + mdb.get(model.MD_INTEGRATION_COUNT, 1)
+
+        return mda
+
+    def subtract_baseline(self, data, md):
+        """
+        Subtract accumulated baselines from the data so that the data only has one.
+        Args:
+            data: the data after the integration of all images
+            md: metadata of the integrated image
+        Returns:
+            data, md: the updated data and metadata after the subtraction of the baseline
+        """
+        baseline = model.MD_BASELINE
```

Comment:
Something went wrong here! `baseline = md[model.MD_BASELINE]`

Please add a BASELINE in one of the test cases. You can use like `data.metadata[model.MD_BASELINE] = 10`.